x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -59,99 +59,86 @@ type InventoryMetricThresholdAllowedActionGroups = ActionGroupIdsOf<
+};
+
+export const createInventoryMetricThresholdExecutor =
+  (libs: InfraBackendLibs) =>
+  async (
+    options: RuleExecutorOptions<
+      InventoryMetricThresholdParams & Record<string, unknown>,
+      InventoryMetricThresholdRuleTypeState,
+      InventoryMetricThresholdAlertState,
+      InventoryMetricThresholdAlertContext,
+      InventoryMetricThresholdAllowedActionGroups,
+      InventoryMetricThresholdAlert
+    >
+  ) => {
+    const {
       services,
       params,
+      startedAt,
       executionId,
       spaceId,
-      startedAt,
       rule: { id: ruleId, tags: ruleTags },
       getTimeRange,
-    }) => {
-      const startTime = Date.now();
+    } = options;
 
-      const { criteria, filterQuery, sourceId = 'default', nodeType, alertOnNoData } = params;
+    const startTime = Date.now();
 
-      if (criteria.length === 0) throw new Error('Cannot execute an alert with 0 conditions');
+    const { criteria, filterQuery, sourceId = 'default', nodeType, alertOnNoData } = params;
 
-      const logger = createScopedLogger(libs.logger, 'inventoryRule', {
-        alertId: ruleId,
-        executionId,
-      });
+    if (criteria.length === 0) throw new Error('Cannot execute an alert with 0 conditions');
+
+    const logger = createScopedLogger(libs.logger, 'inventoryRule', {
+      alertId: ruleId,
+      executionId,
+    });
+
+    const esClient = services.scopedClusterClient.asCurrentUser;
+
+    const { savedObjectsClient, alertFactory: baseAlertFactory, alertsClient } = services;
+
+    if (!alertsClient) {
+      throw new AlertsClientError();
+    }
+
+    if (!params.filterQuery && params.filterQueryText) {
+      try {
+        const { fromKueryExpression } = await import('@kbn/es-query');
+        fromKueryExpression(params.filterQueryText);
+      } catch (e) {
+        logger.error(e.message);
+
+        const actionGroup = FIRED_ACTIONS.id; // Change this to an Error action group when able,
+        const reason = buildInvalidQueryAlertReason(params.filterQueryText);
+
+        const { uuid, start } = alertsClient.report({
+          id: UNGROUPED_FACTORY_KEY,
+          actionGroup,
+        });
+
+        const indexedStartedAt = start ?? startedAt.toISOString();
 
-      const esClient = services.scopedClusterClient.asCurrentUser;
-
-      const {
-        alertWithLifecycle,
-        savedObjectsClient,
-        getAlertStartedDate,
-        getAlertUuid,
-        getAlertByAlertUuid,
-        alertFactory: baseAlertFactory,
-      } = services;
-      const alertFactory: InventoryMetricThresholdAlertFactory = (
-        id,
-        reason,
-        actionGroup,
-        additionalContext,
-        evaluationValues,
-        thresholds
-      ) =>
-        alertWithLifecycle({
-          id,
-          fields: {
+        alertsClient.setAlertData({
+          id: UNGROUPED_FACTORY_KEY,
+          payload: {
             [ALERT_REASON]: reason,
             [ALERT_ACTION_GROUP]: actionGroup,
```

Comment:
we shouldn't need to report this back since it's set by the framework.

Reply:
Done